### PR TITLE
core: limit message queue size

### PR DESCRIPTION
**Pull request by Darredevil**
Reopened PR for #2096

### CODE REVIEW

**Comment by Darredevil** [comment]
@authmillenon @LudwigOrtmann I made another PR as suggested, hopefully this one won't be reverted :-)

**Comment by authmillenon** [comment]
:+1: I'll test it tomorrow :-)

File: core/include/cib.h [hunk 1]
```diff
@@ -17,9 +17,11 @@
  *
  * @author      unknown, propably Kaspar Schleiser <kaspar@schleiser.de>
  */
+
```
Comment (authmillenon) [inline core/include/cib.h]: Introduces trailing whitespace.

**Comment by authmillenon** [comment]
Needs rebase.

**Comment by Darredevil** [comment]
Rebased and ready.

**Comment by Darredevil** [comment]
@authmillenon  @kaspar030 Travis CI approved.

**Comment by Kijewski** [comment]
Looks good. ACK.

**Comment by authmillenon** [comment]
Second ACK needed ;-)!

**Comment by authmillenon** [comment]
I think the most interesting is how the 8-bit platforms and the Cortex-M0 will deal with this. It works on the `samr21-xpro`

```
$ BOARD=samr21-xpro make -C tests/unittests/ tests-core flash test
[…]
2015-04-03 14:50:53,962 - INFO # Connect to serial port /dev/ttyACM0
2015-04-03 14:50:59,055 - INFO # kernel_init(): This is RIOT! (Version: 2014.12-907-g92684-HEAD)
2015-04-03 14:50:59,061 - INFO # kernel_init(): jumping into first task...
2015-04-03 14:51:02,268 - INFO # ...............................................................
2015-04-03 14:51:02,270 - INFO # OK (63 tests)
```

Can somebody confirm for the `arduino-mega2560`? @LudwigOrtmann @N8Fear?

**Comment by LudwigOrtmann** [comment]
Not before Monday.

**Comment by authmillenon** [comment]
Because this is a size optimization, I decided to make a size comparison

```
$ BINDIRBASE=patch-14 make buildtest
[…]
$ git checkout HEAD~1
[…]
$ BINDIRBASE=patch-14~1 make buildtest
[…]
$ OLDBIN=patch-14~1 NEWBIN=patch-14 make info-buildsizes-diff
text	data	bss	dec	BOARD/BINDIRBASE

20	0	0	20	airfy-beacon
15764	148	3916	19828	patch-14~1
15784	148	3916	19848	patch-14

36	0	0	36	arduino-due
16028	152	9192	25372	patch-14~1
16064	152	9192	25408	patch-14

0	0	0	0	arduino-mega2560
15438	942	1130	17510	patch-14~1
15438	942	1130	17510	patch-14

32	0	0	32	avsextrem
60964	2388	95909	159261	patch-14~1
60996	2388	95909	159293	patch-14

36	0	0	36	cc2538dk
15976	148	4596	20720	patch-14~1
16012	148	4596	20756	patch-14

0	0	0	0	chronos
22612	162	3490	26264	patch-14~1
22612	162	3490	26264	patch-14

36	0	0	36	f4vi1
14904	152	9040	24096	patch-14~1
14940	152	9040	24132	patch-14

36	0	0	36	fox
32700	168	9448	42316	patch-14~1
32736	168	9448	42352	patch-14

36	0	0	36	iot-lab_M3
33672	168	9480	43320	patch-14~1
33708	168	9480	43356	patch-14

36	0	0	36	mbed_lpc1768
16264	152	6408	22824	patch-14~1
16300	152	6408	22860	patch-14

0	0	0	0	msb-430
18504	28	1516	20048	patch-14~1
18504	28	1516	20048	patch-14

0	0	0	0	msb-430h
20226	82	3570	23878	patch-14~1
20226	82	3570	23878	patch-14

24	0	0	24	msba2
72172	2396	93921	168489	patch-14~1
72196	2396	93921	168513	patch-14

36	0	0	36	msbiot
23412	212	18948	42572	patch-14~1
23448	212	18948	42608	patch-14

16	0	0	16	native
73890	582	128864	203336	patch-14~1
73906	582	128864	203352	patch-14

20	0	0	20	nucleo-f091
18832	148	2852	21832	patch-14~1
18852	148	2852	21852	patch-14

36	0	0	36	nucleo-f334
15760	148	4348	20256	patch-14~1
15796	148	4348	20292	patch-14

36	0	0	36	nucleo-l1
16412	148	4440	21000	patch-14~1
16448	148	4440	21036	patch-14

36	0	0	36	openmote
15884	148	4596	20628	patch-14~1
15920	148	4596	20664	patch-14

20	0	0	20	pca10000
15792	148	3916	19856	patch-14~1
15812	148	3916	19876	patch-14

20	0	0	20	pca10005
15768	148	3916	19832	patch-14~1
15788	148	3916	19852	patch-14

32	0	0	32	pttu
52660	2332	95965	150957	patch-14~1
52692	2332	95965	150989	patch-14

26	0	0	26	qemu-i386
139887	5528	87668	233083	patch-14~1
139913	5528	87668	233109	patch-14

48	0	0	48	redbee-econotag
71536	2376	20072	93984	patch-14~1
71584	2376	20072	94032	patch-14

84	0	0	84	samr21-xpro
51112	224	8280	59616	patch-14~1
51196	224	8280	59700	patch-14

36	0	0	36	spark-core
16876	148	4440	21464	patch-14~1
16912	148	4440	21500	patch-14

20	0	0	20	stm32f0discovery
18836	148	2852	21836	patch-14~1
18856	148	2852	21856	patch-14

36	0	0	36	stm32f3discovery
14812	148	4372	19332	patch-14~1
14848	148	4372	19368	patch-14

36	0	0	36	stm32f4discovery
15136	152	9048	24336	patch-14~1
15172	152	9048	24372	patch-14

0	0	0	0	telosb
23224	24	3660	26908	patch-14~1
23224	24	3660	26908	patch-14

36	0	0	36	udoo
16028	152	9192	25372	patch-14~1
16064	152	9192	25408	patch-14

0	0	0	0	wsn430-v1_3b
20382	82	3570	24034	patch-14~1
20382	82	3570	24034	patch-14

0	0	0	0	wsn430-v1_4
23332	28	3660	27020	patch-14~1
23332	28	3660	27020	patch-14

20	0	0	20	yunjia-nrf51822
15752	148	3916	19816	patch-14~1
15772	148	3916	19836	patch-14

0	0	0	0	z1
22200	24	3660	25884	patch-14~1
22200	24	3660	25884	patch-14
```

On the 8- and 16-bit platforms we don't win anything, on the 32-bit platforms we loose up to 36 byte. @Kijewski #2096 was your issue (and you ack'd this one already). How shall we proceed?

**Comment by authmillenon** [comment]
(the application above was `examples/default`)

**Comment by LudwigOrtmann** [comment]
same for tests/msg_send_receive with
```diff
diff --git a/core/kernel_init.c b/core/kernel_init.c
index a57f119..a89a907 100644
--- a/core/kernel_init.c
+++ b/core/kernel_init.c
@@ -82,7 +82,7 @@ static char idle_stack[KERNEL_CONF_STACKSIZE_IDLE];
 void kernel_init(void)
 {
     (void) disableIRQ();
-    printf("kernel_init(): This is RIOT! (Version: %s)\n", RIOT_VERSION);
+    //printf("kernel_init(): This is RIOT! (Version: %s)\n", RIOT_VERSION);
```
BTW: the test itself fails (and does so in master as well).

**Comment by authmillenon** [comment]
Is there a specific reason why you removed this line?

**Comment by LudwigOrtmann** [comment]
Yes.

**Comment by authmillenon** [comment]
(would be nice if you can elaborate ;-))

**Comment by LudwigOrtmann** [comment]
To make sure the string length is not the reason for size differences.

**Comment by authmillenon** [comment]
Ah… since I was in HEAD-less mode in git, mine are all empty ;)

**Comment by authmillenon** [comment]
mine == the version strings ;-)

**Comment by authmillenon** [comment]
Just out of curiosity I changed the patch to `s/unsigned int/unsigned short/g`, but the result was unsurprisingly the same.

**Comment by LudwigOrtmann** [comment]
A further example of compiler > developer, or: smaller than memory word data type leads to larger code footprint.

**Comment by Darredevil** [comment]
So what is the conclusion for this patch ? Discard it ?

**Comment by LudwigOrtmann** [comment]
Did you do any research on the problem? What are the memory and execution implications at runtime?

**Comment by OlegHahm** [comment]
> A further example of compiler > developer, or: smaller than memory word data type leads to larger code footprint.

First, this sounds rather like compiler being stupid. Second, why larger code footprint?

**Comment by LudwigOrtmann** [comment]
I didn't look at the assembler, but I guess there are no operations for 16 bit words, so 16 bit arithmetic needs to be "hand optimized".

**Comment by OlegHahm** [comment]
I was referring to @authmillenon's  `but the result was unsurprisingly the same.`

**Comment by authmillenon** [comment]
I wasn't surprised by this, because in the end `uint16_t` is translated to `unsigned short` (or vice-versa).

**Comment by authmillenon** [comment]
(but I had hope, that the compiler does some optimization with the standard types, regardless)

**Comment by LudwigOrtmann** [comment]
@OlegHahm I don't understand your last comment.

**Comment by LudwigOrtmann** [comment]
(I'm reading it in context with your previous comment.)

**Comment by OlegHahm** [comment]
Never mind. I thought the results were the same for `int` and `short`.

**Comment by Darredevil** [comment]
After discussing the issue with @OlegHahm I understand that this patch should best be discarded since it seems it brings no real improvements.
@LudwigOrtmann I would gladly do some further research on the problem if you think it is worth it, but I would ask for some help on how to do the profiling for this. Currently I only have minor experience with Performance Profiling using the tools provided by Visual Studio, but sadly can't use that here.

**Comment by Kijewski** [comment]
To quote myself:

# "But I don't think that the current limit of MAXINT / 2 is reasonable. At least it hurts portability."

**Comment by LudwigOrtmann** [comment]
@Kijewski Can you please spell out what this means for this PR?

**Comment by kaspar030** [comment]
This PR forces the compiler to mask half-words from words on 32bit platforms, thus increasing codesize on those. As this doesn't solve any actual issue, we should discard it.

@Kijewski Isn't ```MAXINT/2``` a portable define?

```cib_init(buf, MAXINT/2)``` should work everywhere?

**Comment by PeterKietzmann** [comment]
So should we close this PR then?

**Comment by LudwigOrtmann** [comment]
yes

**Pull request closed without merging**